packages/cli/src/PublicApi/v1/handlers/executions/executions.service.ts

```diff
@@ -42,58 +34,125 @@ export async function getExecutions(params: {
 		where.id = Not(In(params.excludedExecutionsIds));
 	}
 
-	if (params.status) {
-		where = { ...where, ...getStatusCondition(params.status) };
-	}
 
 	if (params.workflowIds) {
 		where = { ...where, workflowId: In(params.workflowIds) };
 	}
+	if (!params.status) {
+		return Container.get(ExecutionRepository).findMultipleExecutions(
+			{
+				select: [
+					'id',
+					'mode',
+					'retryOf',
+					'retrySuccessId',
+					'startedAt',
+					'stoppedAt',
+					'workflowId',
+					'waitTill',
+					'finished',
+				],
+				where,
+				order: { id: 'DESC' },
+				take: params.limit,
+				relations: ['executionData'],
+			},
+			{
+				includeData: params.includeData,
+				unflattenData: true,
+			},
+		);	
+	}
+	let status_list: ExecutionStatus[];
+	if (typeof(params.status) == "string"){
+		status_list = [params.status];
+	}
+	else {
+		status_list = params.status
+	}
 
-	return Container.get(ExecutionRepository).findMultipleExecutions(
-		{
-			select: [
-				'id',
-				'mode',
-				'retryOf',
-				'retrySuccessId',
-				'startedAt',
-				'stoppedAt',
-				'workflowId',
-				'waitTill',
-				'finished',
-			],
-			where,
-			order: { id: 'DESC' },
-			take: params.limit,
-			relations: ['executionData'],
-		},
-		{
-			includeData: params.includeData,
-			unflattenData: true,
-		},
-	);
+	// findMultipleExecutions can only handle one status at a time.
+	// if this ends up being too slow, see packages/cli/src/executions/executions.service.ts for a different implementation
```

Comment:
Are you sure ?

You removed a function (which was on l.10) where they used `condition.status = In(['error', 'crashed', 'failed']);`

Reply:
Oh... you're right. That makes the code way cleaner. I'll fix that